server/etcdserver/api/membership/cluster.go

```diff
@@ -303,9 +303,27 @@ func (c *RaftCluster) Recover(onSet func(*zap.Logger, *semver.Version)) {
 
 // ValidateConfigurationChange takes a proposed ConfChange and
 // ensures that it is still valid.
-func (c *RaftCluster) ValidateConfigurationChange(cc raftpb.ConfChange) error {
+func (c *RaftCluster) ValidateConfigurationChange(cc raftpb.ConfChange, shouldApplyV3 ShouldApplyV3) error {
 	// TODO: this must be switched to backend as well.
-	membersMap, removedMap := membersFromStore(c.lg, c.v2store)
+	var membersMap map[types.ID]*Member
+	var removedMap map[types.ID]bool
+	if c.v2store != nil {
+		membersMap, removedMap = membersFromStore(c.lg, c.v2store)
+		if err := c.validateConfigurationChange(cc, membersMap, removedMap); err != nil {
+			return err
+		}
+	}
+
+	if c.be != nil && shouldApplyV3 {
+		membersMap, removedMap = c.be.MustReadMembersFromBackend()
+		if err := c.validateConfigurationChange(cc, membersMap, removedMap); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
```

Comment:
Can: 
```
membersMap map[types.ID]*Member
removedMap map[types.ID]bool
```
be properties of RaftClusters... as mutable arguments to functions are not readable. 

Like: 
  `currentReplayV2MembersMap` & `currentReplayV2removedMap...`

Reply:
Not sure I understand, do you want to propose to also validate the RaftCluster.membersMap and RaftCluster.removedMap? Note; before this PR don't validate them, we only validate v2 store.